### PR TITLE
Added click event type support

### PR DESCRIPTION
**Pull request by Wokstym**
Types from docs, i left doing type for event for later as it is humongous, 
https://visjs.github.io/vis-network/docs/network/

Also I've set `items` type as unknown because I couldn't get values which are in docs, i am not sure what `labelId` is
image

### CODE REVIEW

File: src/EventTypes.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+export type GraphEvents = Partial<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | Record<UnsupportedTypeNetworkEvents, (params?: any) => void>
```
Comment (WikKam) [inline src/EventTypes.ts]: maybw params?: unknown

File: src/EventTypes.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  | 'initRedraw'
+  | 'beforeDrawing'
+  | 'afterDrawing'
+  | 'animationFinished'
+  | 'configChange';
+
+export interface BaseClickData {
+  nodes: string[];
+  edges: string[];
+  event: unknown; // TODO add event type
```
Comment (WikKam) [inline src/EventTypes.ts:35]: what about that? Why this is unknown also?

Comment (WikKam) [reply src/EventTypes.ts:35]: this should be MouseEvent or its react equivalent probably

Comment (Wokstym) [reply src/EventTypes.ts:35]: I have trouble finding some existing type, maybe it is a custom from vis js?

**Comment by ZachHaber** [comment]
There's a few issues with the types as is. I'll post a change with slightly better typing.
Mostly we should be using intersection types to combine things instead of unions, and I set it up so we don't have to set up our list of unsupported ones directly either.

File: src/EventTypes.ts (hunk 2 above)
Comment (ZachHaber) [reply src/EventTypes.ts:35]: We should try and add the properties that exist... 
```
type SubEvent= {
    "pointers": unknown[],
    "changedPointers": PointerEvent[
    ],
    "pointerType": string,// or a union probably?
    "srcEvent": PointerEvent,
    "isFirst": boolean,
    "isFinal": boolean,
    "eventType": number,// enum?
    "center": {
        "x": number,
        "y": number
    },
    "timeStamp": number,
    "deltaTime": number,
    "angle": number,
    "distance": number,
    "deltaX": number,
    "deltaY": number,
    "offsetDirection": number,
    "overallVelocityX": number,
    "overallVelocityY": number,
    "overallVelocity": number,
    "scale": number,
    "rotation": number,
    "maxPointers": number,
    "velocity": number,
    "velocityX": number,
    "velocityY": number,
    "direction": number,
    "target": HTMLCanvasElement,
    "type": "doubletap",
    "tapCount": number
}
```

Comment (Wokstym) [reply src/EventTypes.ts:35]: Lets do it separately, I will include this in the issue

**Review by Wokstym — COMMENTED** [review]
Looks good for me

**Pull request merged**